pkg/virt-controller/watch/workload-updater/workload-updater_test.go

```diff
@@ -407,35 +467,56 @@ var _ = Describe("Workload Updater", func() {
 
 	Context("LiveUpdate features", func() {
 		It("VMI needs to be migrated when memory hotplug is requested", func() {
-			vmi := api.NewMinimalVMI("testvm")
-
 			condition := v1.VirtualMachineInstanceCondition{
 				Type:   v1.VirtualMachineInstanceMemoryChange,
 				Status: k8sv1.ConditionTrue,
 			}
+			vmi := libvmi.New(
+				libvmi.WithName("testvm"),
+				libvmistatus.WithStatus(
+					libvmistatus.New(libvmistatus.WithCondition(condition)),
+				),
+			)
+
 			virtcontroller.NewVirtualMachineInstanceConditionManager().UpdateCondition(vmi, &condition)
 
 			Expect(controller.doesRequireMigration(vmi)).To(BeTrue())
 		})
 	})
 
 	Context("Abort changes due to an automated live update", func() {
-		createVM := func(annotation, hasChangeCondition bool) *v1.VirtualMachineInstance {
-			vmi := api.NewMinimalVMI("testvm")
-			vmi.Namespace = k8sv1.NamespaceDefault
-			vmi.Status.Phase = v1.Running
-			vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
-				Type: v1.VirtualMachineInstanceIsMigratable, Status: k8sv1.ConditionTrue})
+		const (
+			withAnnotation               = true
+			withoutAnnotation            = false
+			withMemoryChangeCondition    = true
+			withoutMemoryChangeCondition = false
+		)
+		createVM := func(hasAbortionAnnotation, hasChangeCondition bool) *v1.VirtualMachineInstance {
+			statusOpts := []libvmistatus.StatusOption{
+				libvmistatus.WithPhase(v1.Running),
+				libvmistatus.WithCondition(v1.VirtualMachineInstanceCondition{
+					Type: v1.VirtualMachineInstanceIsMigratable, Status: k8sv1.ConditionTrue},
+				),
+			}
 			if hasChangeCondition {
-				vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
-					Type:   v1.VirtualMachineInstanceMemoryChange,
-					Status: k8sv1.ConditionTrue,
-				})
+				statusOpts = append(statusOpts,
+					libvmistatus.WithCondition(
+						v1.VirtualMachineInstanceCondition{
+							Type:   v1.VirtualMachineInstanceMemoryChange,
+							Status: k8sv1.ConditionTrue,
+						},
+					),
+				)
 			}
-			if annotation {
-				vmi.ObjectMeta.Annotations = make(map[string]string)
-				vmi.ObjectMeta.Annotations[v1.WorkloadUpdateMigrationAbortionAnnotation] = ""
+			opts := []libvmi.Option{
+				libvmi.WithName("testvm"),
```

Comment:
Is this a must? Can it use the randomized name?

Reply:
some WU tests creates several vmi with different names and play with them with which one is going to be updated or not.
Generally, for production purpose, I think `WithName` option is quite useful.